common/src/main/scala/explore/utils/CloneListView.scala

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.utils
+
+import cats.Monad
+import crystal.ViewF
+import crystal.ViewListF
+import japgolly.scalajs.react.util.DefaultEffects.{Sync => DefaultS}
+
+// Turn a ViewList into a View, assuming all returned objects are clones.
+// and therefore just returning the head upon get.
```

Comment:
A `ViewList` can be seen as a `View`+`Traversal`, where the `get` returns a `List[A]` but the `mod`/`set` operates on `A` (not on `List[A]`), thus affecting all the elements the same way.

`CloneListView` assumes all the elements in the `List` are the same (clones of each other), and therefore exposes regular `View[A]` semantics by returning the `head` upon `get` (but it could return any element from the list, since they are all clones).

I found this very convenient when working with groupings. For example, when grouping observations by constraint set, we can create a `ViewList` into the `ConstraintSet` of the observations in the same group. The `ConstraintSet` of all these observations will be clones of each other (since that was actually the criterion for grouping them). We can pass a `CloneListView[ConstraintSet]` into a component that expects a `View[ConstraintSet]` and all the changes will be applied to all the observations in the group.

Reply:
Cool!